aks/postgres/data.tf

```diff
@@ -31,3 +31,9 @@ data "azurerm_monitor_action_group" "main" {
   name                = "${var.azure_resource_prefix}-${var.service_name}"
   resource_group_name = data.azurerm_resource_group.monitoring[0].name
 }
+
+data "azurerm_monitor_diagnostic_categories" "postgres" {
+  count = var.azure_enable_backing_services ? 1 : 0
```

Comment:
```suggestion
  count = local.azure_enable_monitoring ? 1 : 0
```

We should use this variable instead, which is true if `use_azure` is `true` and `azure_enable_monitoring` is `true`.

Reply:
Thats specific to monitoring and alerting this setting is specific to diagnostics settings and has a depenedency on backing services to be enabled and only needs one condition to be true. Ideally the setting only should apply to production postgres instance and not for lower environments.

Reply:
Yeah, `use_azure` will be set to `true` if we're deploying the database to Azure rather than a Kubernetes service, and `azure_enable_monitoring` is typically only set to `true` in production environments.

If we think log analytics needs to be separate to monitoring, then I think we should introduce an `azure_enable_logging` variable.

Reply:
resolved